pkg/server/chaosd/redis.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosd
+
+import (
+	"os/exec"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/pingcap/errors"
+
+	"github.com/chaos-mesh/chaosd/pkg/core"
+)
+
+type redisAttack struct{}
+
+var RedisAttack AttackType = redisAttack{}
+
+const (
+	STATUSOK = "OK"
+)
+
+func (redisAttack) Attack(options core.AttackConfig, env Environment) error {
+	attack := options.(*core.RedisCommand)
+
+	cli := redis.NewClient(&redis.Options{
+		Addr:     attack.Addr,
+		Password: attack.Password,
+	})
+	_, err := cli.Ping(cli.Context()).Result()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	switch attack.Action {
+	case core.RedisSentinelRestartAction:
+		err := env.Chaos.shutdownSentinelServer(attack, cli)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		return env.Chaos.recoverSentinelStop(attack)
```

Comment:
seems that `restart` only does this one more thing than `stop`,  I think we can only remain one action.

Reply:
Do you mean remove `sentinel stop`? I'm not sure that's a good decision, because `stop` and 'restart` are two separate functions in cloud chaos.

Reply:
It's ok to keep the `sentinel stop`